### ISSUE TITLE
add support for the ServiceApplicationClient oauthlib.oauth2 client

### ISSUE THREAD

**Issue by paulie4** [SalesforceFoundation/salesforce-requests-oauthlib#7]
I'd like to be able to use the `salesforce_requests_oauthlib` module with a JWT bearer token (via the `ServiceApplicationClient` class in `oauthlib.oauth2`), but the `client` parameter that the super-class's `__init__()` accepts is not available from `SalesforceOAuth2Session.__init__()`. Can we add that? Of course, that also means the code can no longer expect to have a refresh_token, since the [OAuth 2.0 JWT Bearer Token Flow](https://help.salesforce.com/articleView?id=remoteaccess_oauth_jwt_flow.htm) does not use refresh_tokens.

**Comment by paulie4** [comment]
The pull request I submitted allows someone to create an `oauthlib.oauth2` client and then pass it like this:
```
from oauthlib.oauth2 import ServiceApplicationClient
client = ServiceApplicationClient(clientid, certkey, user, clientid, audience)
sfOAuthSess = SalesforceOAuth2Session(clientid, None, user, oauth2client=client)
```

**Comment by adamlincoln** [comment]
Closed by PR #6 , thanks @paulie4 .

**Comment by adamlincoln** [comment]
Available in release 0.1.4 on pypi.
